### PR TITLE
NODE-1828: cleanup async resources on close

### PR DESCRIPTION
**Pull request by mbroadst**
The native component to https://github.com/mongodb-js/mongodb-core/pull/417

### CODE REVIEW

File: lib/topologies/topology_base.js [hunk 1]
```diff
@@ -390,9 +390,12 @@ class TopologyBase extends EventEmitter {
...
-    });
+    this.s.coreTopology.destroy(
+      {
+        force: typeof forceClosed === 'boolean' ? forceClosed : false
+      },
+      callback
+    );
 
     // We need to wash out all stored processes
     if (forceClosed === true) {
```
Comment (daprahamian) [inline lib/topologies/topology_base.js]: Should this be part of the callback after ` this.s.coreTopology.destroy(`?

Comment (mbroadst) [reply lib/topologies/topology_base.js]: My thought was that if you are doing a force close, then flushing these immediately should be expected. If its not a force, then the async close will wait for all buffered operations to complete. Maybe this should go _before_ the call to `destroy`?

Comment (daprahamian) [reply lib/topologies/topology_base.js]: Yeah, that might be the best route. does that affect the results of any of our tests?

Comment (mbroadst) [reply lib/topologies/topology_base.js]: I just tested again with this change, no breakage

**Pull request merged**